tests/publiccloud/storage_perf.pm

```diff
@@ -121,7 +130,8 @@ sub run {
     select_serial_terminal();
 
     my $provider = $self->provider_factory();
-    my $instance = $provider->create_instance(use_extra_disk => {size => $disk_size, type => $disk_type});
+    my $use_extra_drive_variable = $use_nvme ? {} : {size => $disk_size, type => $disk_type};
```

Comment:
Is this really needed? Shouldn't the nvme-related tests simply not have those settings and be done?

I'm saying it because ignoring settings without a warning or error as a safeguard is in general very confusing for the users. It is unexpected.

Reply:
mmm another way to write it would be 

```
my $instance;
if ( $use_nvme) {
$instance = $provider->create_instance(use_extra_disk => $use_extra_drive_variable);
}
else {
$instance = $provider->create_instance(use_extra_disk => $use_extra_drive_variable);
}
```

I thought mine one looks better but if you are against I can switch to this one ...